node/primitives/src/lib.rs

```diff
@@ -87,5 +90,22 @@ pub type ProtocolId = [u8; 4];
 /// A succinct representation of a peer's view. This consists of a bounded amount of chain heads.
 ///
 /// Up to `N` (5?) chain heads.
-#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+#[derive(Default, Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct View(pub Vec<Hash>);
+
+impl View {
+	/// Returns an iterator of the hashes present in `Self` but not in `other`.
+	pub fn difference<'a>(&'a self, other: &'a View) -> impl Iterator<Item = Hash> + 'a {
+		self.0.iter().cloned().filter(move |h| !other.contains(h))
```

Comment:
Is the `cloned()` necessary here? Anything impeding returning a `impl Iterator<Item=&'a Hash>` instead? It can then be decided in a case by case if cloning is necessary (and from what I can see in most cases it's not necessary)